File: patch-01/backend/inspect_schemas.py
import mysql.connector

try:
    conn = mysql.connector.connect(
        host="127.0.0.1",
        user="root",
        password="",
        database="iwpaint_optimized"
    )
    cursor = conn.cursor()
    tables = ['sales_transactions', 'products', 'customers', 'salesmen', 'supervisors', 'salesman_targets']
    for t in tables:
        print(f"\n--- {t} ---")
        try:
            cursor.execute(f"DESCRIBE {t}")
            for row in cursor.fetchall():
                print(f"{row[0]} ({row[1]})")
        except Exception as e:
            print(f"Error describing {t}: {e}")
    conn.close()
except Exception as e:
    print(f"Error connecting: {e}")

File: patch-01/backend/inspect_schemas.js
const mysql = require('mysql2/promise');
async function run() {
  const pool = mysql.createPool({ host:'127.0.0.1', user:'root', password:'', database:'iwpaint_optimized'});
  const tables = ['sales_transactions', 'products', 'customers', 'salesmen', 'supervisors', 'salesman_targets'];
  for (const t of tables) {
    console.log(`\n--- ${t} ---`);
    try {
      const [cols] = await pool.query(`DESCRIBE ${t}`);
      console.log(cols.map(c => `${c.Field} (${c.Type})`).join(', '));
    } catch(e) {
      console.log(`Error describing ${t}: ${e.message}`);
    }
  }
  process.exit(0);
}
run();
